src/main/java/it/pagopa/ecommerce/commons/domain/BIN.java

```diff
@@ -0,0 +1,32 @@
+package it.pagopa.ecommerce.commons.domain;
+
+import java.util.Objects;
+
+/**
+ * Domain class representing a credit/debit card BIN (bank identification
+ * number)
+ *
+ * @param value the card bin
+ */
+public record BIN(String value) {
+    /**
+     * BIN format used to check for valid values
+     */
+    public static final String BIN_FORMAT = "\\d{4}\\d?\\d?";
```

Comment:
As for now we take the first 8 pan digits for the bin
```suggestion
    public static final String BIN_FORMAT = "\\d{4}\\d{0,4}";
```

Reply:
Fixed in 7cee3293